packages/teleport-types/src/generators.ts

```diff
@@ -111,9 +112,12 @@ export interface ProjectStrategy {
     path: string[]
     options?: {
       createFolderForEachComponent?: boolean
-      customComponentFileName?: string // only used when createFolderForEachComponent is true
-      customStyleFileName?: string
-      customTemplateFileName?: string
+      customComponentFileName?: (name?: string) => string // only used when createFolderForEachComponent is true
+      customStyleFileName?: (name?: string) => string
+      customTemplateFileName?: (name?: string) => string
+      module?: {
```

Comment:
I would name this `moduleGenerator` and place it next to `generator` and `path` since it's an important part of the  strategy